### PR TITLE
add support for JuliaLang/llvm-project

### PR DESCRIPTION
**Pull request by vchuravy**
https://github.com/llvm/llvm-project/compare/release/13.x...JuliaLang:julia/release/13.x

### CODE REVIEW

**Comment by vchuravy** [comment]
@vtjnash I moved to `git-external` as you mentioned. 

- [ ] Optional: Support building against RV again
- [ ] BUILD_CUSTOM_LIBCXX

File: deps/llvm.mk [hunk 1]
```diff
@@ -358,103 +282,56 @@ LIBCXX_DEPENDENCY := $(build_libdir)/libc++abi.so.1.0 $(build_libdir)/libc++.so.
...
+# # ifeq ($(USE_RV),1)
+# # 	git clone -b $(LLVM_RV_GIT_VER) $(LLVM_RV_GIT_URL) $(LLVM_MONOSRC_DIR)/rv
+# # 	(cd $(LLVM_MONOSRC_DIR)/rv && \
+# # 		git submodule update --init) || true
+# # endif
+# 	# touch some extra files to ensure bisect works pretty well
+# 	touch -c $(SRCCACHE)/$(LLVM_SRC_DIR).extracted
+# 	touch -c $(LLVM_SRC_DIR)$(SRCCACHE)/$(LLVM_SRC_DIR)/configure
+# 	touch -c $(LLVM_SRC_DIR)$(SRCCACHE)/$(LLVM_SRC_DIR)/CMakeLists.txt
+# 	echo 1 > $@
```
Comment (vtjnash) [inline deps/llvm.mk:315]: delete these lines?

File: deps/llvm.mk [hunk 2]
```diff
@@ -585,18 +431,19 @@ clean-llvm: clean-libcxx clean-libcxxabi
...
 	-$(MAKE) -C $(LLVM_BUILDDIR_withtype) clean
 
-distclean-llvm: distclean-libcxx distclean-libcxxabi
-	-rm -rf $(LLVM_TAR) $(LLVM_CLANG_TAR) \
-		$(LLVM_COMPILER_RT_TAR) $(LLVM_LIBCXX_TAR) $(LLVM_LLDB_TAR) \
-		$(LLVM_SRC_DIR) $(LLVM_BUILDDIR_withtype)
+# distclean-llvm: distclean-libcxx distclean-libcxxabi
+# 	-rm -rf $(LLVM_TAR) $(LLVM_CLANG_TAR) \
+# 		$(LLVM_COMPILER_RT_TAR) $(LLVM_LIBCXX_TAR) $(LLVM_LLDB_TAR) \
+# 		$(LLVM_SRC_DIR) $(LLVM_BUILDDIR_withtype)
```
Comment (vtjnash) [inline deps/llvm.mk:437]: delete?

File: deps/llvm.mk [hunk 3]
```diff
@@ -585,18 +431,19 @@ clean-llvm: clean-libcxx clean-libcxxabi
...
-get-llvm: $(LLVM_TAR) $(LLVM_CLANG_TAR) $(LLVM_COMPILER_RT_TAR) $(LLVM_LIBCXX_TAR) $(LLVM_LLDB_TAR)
-else
-get-llvm: $(LLVM_SRC_DIR)/source-extracted
-endif
-extract-llvm: $(LLVM_SRC_DIR)/source-extracted
+# ifneq ($(LLVM_VER),svn)
+# get-llvm: $(LLVM_TAR) $(LLVM_CLANG_TAR) $(LLVM_COMPILER_RT_TAR) $(LLVM_LIBCXX_TAR) $(LLVM_LLDB_TAR)
+# else
+# get-llvm: $(LLVM_SRC_DIR)/source-extracted
+# endif
```
Comment (vtjnash) [inline deps/llvm.mk:444]: delete?

**Review by vtjnash — COMMENTED** [review]
great to be deleting so many lines, remove the patches themselves too?

File: deps/llvm.version [hunk 4]
```diff
@@ -0,0 +1,2 @@
+LLVM_BRANCH=julia-12.0.1-3
+LLVM_SHA1=8f6e49e8a3c0f31a55c342a2a41dda630508c5eb
```
Comment (vtjnash) [inline deps/llvm.version:2]:
```suggestion
LLVM_SHA1= julia-12.0.1-3
```
It is actually a stable commit reference, so the tag is a preferable name

**Review by vtjnash — APPROVED** [review]

**Comment by fxcoudert** [comment]
This caused a regression on master: https://github.com/JuliaLang/julia/issues/44499

**Pull request merged**